sympy/series/approximants.py

```diff
@@ -101,3 +108,282 @@ def approximants(l, X=Symbol('x'), simplify=False):
         else:
             yield out
     return
+
+
+@public
+def pade_approximants(
+    f: Poly, x: Expr, order: int
+) -> Generator[tuple[Poly, Poly], None, None]:
+    """
+    Returns all pade approximants of the expression `f` of the desired order.
+
+    Description
+    ===========
+
+    Computes all pade approximants `f` of order `[m/n]` with `m + n =` order.
+
+    Parameters
+    ==========
+
+    f : Poly
+        The polynomial to approximate
+    x : Expr
+        The variable of p
+    order : int
+        The desired order of the pade approximants
+
+    Returns
+    =======
+
+    pade approximants : Generator[tuple[Poly, Poly], None, None]
+        Generator for (numerator, denominator) pairs of polynomials
+        representing the numerator and denominators of the approximants.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x
+    >>> from sympy import Poly
+    >>> from sympy.series.approximants import pade_approximants
+
+    >>> exp_taylor_poly = Poly(1 + x + x**2/2 + x**3/6 + x**4/24, x, domain='QQ')
+    >>> pade_exp = pade_approximants(exp_taylor_poly, x, 2)
+    >>> for num, denom in pade_exp: print(num, denom)
+    Poly(1/2*x**2 + x + 1, x, domain='QQ') Poly(1, x, domain='QQ')
+    Poly(2*x + 4, x, domain='QQ') Poly(-2*x + 4, x, domain='QQ')
+    Poly(1, x, domain='QQ') Poly(1/2*x**2 - x + 1, x, domain='QQ')
+
+    To get pade approximants, divide the numerators by the denominators
+
+    >>> sin_taylor_poly = Poly(x - x**3/6 + x**5/120, x, domain='QQ')
+    >>> pade_sin = pade_approximants(sin_taylor_poly, x, 5)
+    >>> for num, denom in pade_sin: print(num/denom)
+    x**5/120 - x**3/6 + x
+    -(20*x**4 - 120*x**2)/(120*x)
+    (-7*x**3/60 + x)/(x**2/20 + 1)
+    360*x**2/(7*(60*x**3/7 + 360*x/7))
+    x/(7*x**4/360 + x**2/6 + 1)
+
+    See also
+    ========
+
+    sympy.series.approximants.approximants
+    sympy.series.approximants.pade_approximant_gcdex
+    sympy.series.approximants.pade_approximant
+    """
+
+    if order < 0:
+        raise ValueError("'order' must be a non-negative integer")
+
+    # truncate f so f.degree() <= order
+    if f.degree() > order:
+        coefficients = f.all_coeffs()
+        f = Poly.from_list(coefficients[-(order + 1):], gens=x)
+
+    remainder_monomial = Poly(x ** (order + 1), x)
+
+    yield f, f.one
+
+    euclidean_algorithm_steps = gcdex_steps(remainder_monomial, f)
+
+    for _, t, r in euclidean_algorithm_steps:
+        yield r, t
+
+
+@public
+def pade_approximant_gcdex(
+    f: Poly, x: Expr, m: int, n: int | None = None
```

Comment:
The argument `x` is redundant here if `f` is a Poly. Instead the function can just check that the `Poly` is univariate.